src/software/ai/hl/stp/evaluation/possession.cpp

```diff
@@ -49,17 +49,16 @@ namespace Evaluation
 
     bool teamHasPossession(const Team &team, const Ball &ball)
     {
-        for (Robot robot : team.getAllRobots())
+        for (const Robot &robot : team.getAllRobots())
         {
             std::vector<Timestamp> robot_history_timestamps =
                 robot.getPreviousTimestamps();
 
             int i = 0;
 
             // Check that the robot has had possession of the ball recently.
-            while (i < robot_history_timestamps.size() &&
```

Comment:
Why did we remove this size check at competition? It's likely the cause of the possession unit test segfaulting.

Reply:
Not sure. I thought I re-added that size-check somewhere but apparently not? I found the same segfault at some point. Should be fine to re-add